packages/web-app-files/tests/unit/components/SideBar/SideBar.spec.js

```diff
@@ -42,51 +45,327 @@ function createWrapper({ item, selectedItems, mocks }) {
+  return mount(SideBar, getMountOptions(options))
+}
+
+function createWrapper(options) {
+  return shallowMount(SideBar, getMountOptions(options))
+}
+
+const panelSelectors = {
+  Details: '#sidebar-panel-details-item',
+  Actions: '#sidebar-panel-actions-item',
+  People: '#sidebar-panel-sharing-item'
 }
 
+const selectors = {
+  panelBody: '.sidebar-panel__body',
+  actionButton: '#sidebar-panel-actions-item-select',
+  peopleButton: '#sidebar-panel-sharing-item-select'
+}
+
+const samplePanels = [
+  () => {
+    return {
+      app: 'details-item',
+      get enabled() {
+        return true
+      },
+      icon: 'info_outline',
+      component: getSideBarComponent('Details'),
+      default: () => true
+    }
+  },
+  () => {
+    return {
+      app: 'actions-item',
+      get enabled() {
+        return true
+      },
+      icon: 'slideshow',
+      component: getSideBarComponent('Actions')
+    }
+  },
+  () => {
+    return {
+      app: 'sharing-item',
+      get enabled() {
+        return true
+      },
+      icon: 'group',
+      component: getSideBarComponent('People')
+    }
+  }
+]
+
 describe('SideBar', () => {
-  it('fetches file info if 1 item is selected', () => {
-    const mockFileInfo = jest.fn()
-    mockFileInfo.mockReturnValueOnce(Files['/'][1])
-
-    createWrapper({
-      item: simpleOwnFolder,
-      selectedItems: [simpleOwnFolder],
-      mocks: { $client: { files: { fileInfo: mockFileInfo } } }
+  describe('Fetching the fileinfo of selected file', () => {
+    it('fetches file info if 1 item is selected', () => {
+      const mockFileInfo = jest.fn()
+      mockFileInfo.mockReturnValueOnce(Files['/'][1])
+
+      createWrapper({
+        item: simpleOwnFolder,
+        selectedItems: [simpleOwnFolder],
+        mocks: { $client: { files: { fileInfo: mockFileInfo } } }
+      })
+
+      expect(mockFileInfo).toHaveBeenCalledTimes(1)
     })
 
-    expect(mockFileInfo).toHaveBeenCalledTimes(1)
+    it('does not fetch file info if multiple items are selected', () => {
+      const mockFileInfo = jest.fn()
+
+      createWrapper({
+        item: simpleOwnFolder,
+        selectedItems: [simpleOwnFolder, simpleOwnFolder],
+        mocks: { $client: { files: { fileInfo: mockFileInfo } } }
+      })
+
+      expect(mockFileInfo).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('When on root folder', () => {
+    it('should show no selection panel', async () => {
+      const item = getResource({ filename: '', type: 'folder' })
+
+      const wrapper = createWrapper({
+        item,
+        selectedItems: [],
+        mocks: {
+          $client: {
+            files: {
+              fileInfo: () =>
+                getFileResponse({
+                  filename: '',
+                  extension: '',
+                  type: 'folder'
+                })
+            }
+          }
+        }
+      })
+
+      await flushPromises()
+
+      expect(wrapper).toMatchSnapshot()
```

Comment:
I didn't expect a snapshot test here. The test only says that it expects the `no selection` panel to be present. Please only test that the panel is present. Snapshot test is testing too much.